Source/WebKit/Shared/Cocoa/CoreIPCContacts.h

```diff
@@ -89,6 +90,49 @@ class CoreIPCCNPhoneNumber {
     String m_countryCode;
 };
 
+class CoreIPCCNContact {
+public:
+    CoreIPCCNContact(CNContact *);
+
+    RetainPtr<id> toID() const;
+
+    static bool isValidCNContactType(NSInteger);
+
+private:
+    friend struct IPC::ArgumentCoder<CoreIPCCNContact, void>;
+    CoreIPCCNContact()
+    {
+    }
+
+    String m_identifier;
+    NSInteger m_contactType;
+
+    std::optional<String> m_namePrefix;
+    std::optional<String> m_givenName;
+    std::optional<String> m_middleName;
+    std::optional<String> m_familyName;
+    std::optional<String> m_previousFamilyName;
+    std::optional<String> m_nameSuffix;
+    std::optional<String> m_nickname;
+    std::optional<String> m_organizationName;
+    std::optional<String> m_departmentName;
+    std::optional<String> m_jobTitle;
+    std::optional<String> m_phoneticGivenName;
+    std::optional<String> m_phoneticMiddleName;
+    std::optional<String> m_phoneticFamilyName;
+    std::optional<String> m_phoneticOrganizationName;
+    std::optional<String> m_note;
+
+    std::optional<CoreIPCDateComponents> m_birthday;
+    std::optional<CoreIPCDateComponents> m_nonGregorianBirthday;
+
+    Vector<std::tuple<String, String, CoreIPCDateComponents>> m_dates;
```

Comment:
I don't think we should use tuple.  Give the pieces a name with a nested struct.